source/postcard-schema/src/schema/mod.rs

```diff
@@ -104,75 +90,78 @@ pub enum DataModelType {
-    TupleStruct(&'static [&'static NamedType]),
+    Tuple(&'static [&'static Self]),
 
     /// The "Map" Serde Data Model Type
     Map {
         /// The map "Key" type
-        key: &'static NamedType,
+        key: &'static Self,
         /// The map "Value" type
-        val: &'static NamedType,
+        val: &'static Self,
     },
 
-    /// The "Struct" Serde Data Model Type
-    Struct(&'static [&'static NamedValue]),
+    /// One of the struct Serde Data Model types
+    Struct {
+        /// The name of this struct
+        name: &'static str,
+        /// The data contained in this struct
+        data: Data,
+    },
 
     /// The "Enum" Serde Data Model Type (which contains any of the "Variant" types)
-    Enum(&'static [&'static NamedVariant]),
+    Enum {
+        /// The name of this struct
+        name: &'static str,
+        /// The variants contained in this enum
+        variants: &'static [&'static Variant],
+    },
 
-    /// A NamedType/OwnedNamedType
+    /// A [`DataModelType`]/[`OwnedDataModelType`](owned::OwnedDataModelType)
     Schema,
 }
 
-/// This is similar to [`DataModelType`], however it only contains the potential Data Model Types
-/// used as variants of an `enum`.
+/// The contents of a struct or enum variant.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize)]
-pub enum DataModelVariant {
-    /// The "unit variant" Serde Data Model Type
-    UnitVariant,
-    /// The "newtype variant" Serde Data Model Type
-    NewtypeVariant(&'static NamedType),
-    /// The "Tuple Variant" Serde Data Model Type
-    TupleVariant(&'static [&'static NamedType]),
-    /// The "Struct Variant" Serde Data Model Type
-    StructVariant(&'static [&'static NamedValue]),
+pub enum Data {
+    /// The "Unit Struct" or "Unit Variant" Serde Data Model Type
+    Unit,
+
+    /// The "Newtype Struct" or "Newtype Variant" Serde Data Model Type
+    Newtype(&'static DataModelType),
+
+    /// The "Tuple Struct" or "Tuple Variant" Serde Data Model Type
+    Tuple(&'static [&'static DataModelType]),
+
+    /// The "Struct" or "Struct Variant" Serde Data Model Type
+    Struct(&'static [&'static NamedField]),
 }
 
 /// This represents a named struct field.
 ///
 /// For example, in `struct Ex { a: u32 }` the field `a` would be reflected as
-/// `NamedValue { name: "a", ty: DataModelType::U32 }`.
+/// `NamedField { name: "a", ty: DataModelType::U32 }`.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize)]
-pub struct NamedValue {
-    /// The name of this value
+pub struct NamedField {
+    /// The name of this field
     pub name: &'static str,
-    /// The type of this value
-    pub ty: &'static NamedType,
+    /// The type of this field
+    pub ty: &'static DataModelType,
 }
 
-/// An enum variant with a name, e.g. `T::Bar(...)`
+/// An enum variant e.g. `T::Bar(...)`
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize)]
-pub struct NamedVariant {
+pub struct Variant {
```

Comment:
All variants have names so I thought it made sense to remove the `Named` prefix, can revert